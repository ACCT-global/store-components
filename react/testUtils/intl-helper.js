import React from 'react'
import * as reactTestingLibrary from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import defaultStrings from '../../messages/en-US.json'
const Context = React.createContext({})

export const render = (node, options) => {
  const rendered = reactTestingLibrary.render(
    <IntlProvider messages={defaultStrings} locale="en-US">
      {node}
    </IntlProvider>,
    options
  )

  return {
    ...rendered,
    rerender: newUi =>
      render(newUi, {
        container: rendered.container,
        baseElement: rendered.baseElement,
      }),
  }
}

// re-export everything
export default reactTestingLibrary
