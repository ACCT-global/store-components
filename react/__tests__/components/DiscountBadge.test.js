import React from 'react'
import { renderWithIntl } from 'intl-helper'

import DiscountBadge from '../../DiscountBadge'

describe('<DiscountBadge />', () => {
  function renderComponent(customProps = {}) {
    const props = {
      listPrice: 100,
      sellingPrice: 90,
      ...customProps,
    }
    const component = <DiscountBadge {...props}>Test</DiscountBadge>

    return renderWithIntl(component)
  }

  it('should be able to mount and not break', () => {
    const comp = renderComponent()
    expect(comp).toBeTruthy()
  })

  it('should match snapshot without label', () => {
    const comp = renderComponent()
    expect(comp).toMatchSnapshot()
  })

  it('should match snapshot with label', () => {
    const comp = renderComponent({ label: 'LABEL' })
    expect(comp).toMatchSnapshot()
  })
})
