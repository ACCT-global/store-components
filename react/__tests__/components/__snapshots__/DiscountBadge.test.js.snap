// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscountBadge /> should match snapshot with label 1`] = `
<div
  class="discountContainer relative dib w-100"
>
  <div
    class="t-mini white absolute right-0 pv2 ph3 bg-emphasis"
  >
    <span>
      10%
    </span>
     LABEL
  </div>
  Test
</div>
`;

exports[`<DiscountBadge /> should match snapshot without label 1`] = `
<div
  class="discountContainer relative dib w-100"
>
  <div
    class="t-mini white absolute right-0 pv2 ph3 bg-emphasis"
  >
    -
    <span>
      10%
    </span>
     
  </div>
  Test
</div>
`;
